libgnucash/engine/Account.cpp

```diff
@@ -5534,29 +5535,23 @@ gnc_account_delete_map_entry (Account *acc, char *full_category, gboolean empty)
 
 /*******************************************************************************/
 
-static gchar *
-look_for_old_separator_descendants (Account *root, gchar const *full_name, const gchar *separator)
+static std::string
+look_for_old_separator_descendants (Account *root, std::string const & full_name, const gchar *separator)
 {
     GList *top_accounts, *ptr;
     gint   found_len = 0;
     gchar  found_sep;
-    gchar * new_name = nullptr;
-
     top_accounts = gnc_account_get_descendants (root);
-
-    PINFO("Incoming full_name is '%s', current separator is '%s'", full_name, separator);
-
+    PINFO("Incoming full_name is '%s', current separator is '%s'", full_name.c_str (), separator);
     /* Go through list of top level accounts */
     for (ptr = top_accounts; ptr; ptr = g_list_next (ptr))
     {
         const gchar *name = xaccAccountGetName (static_cast <Account const *> (ptr->data));
-
         // we are looking for the longest top level account that matches
-        if (g_str_has_prefix (full_name, name))
+        if (g_str_has_prefix (full_name.c_str (), name))
```

Comment:
Come on. If you're going to introduce a C++ feature, **use it!** Possibilities:
```
if (full_name.find(name) == full_name.begin())
```
or
```
auto const name == std::string(xaccAccountGetName(static_cast<Account const*>(ptr-data));
if (full_name.substr(0, name.length()) == name)
```
Not a blocker for merging, just a plea to write C++ when you're changing code.

Reply:
Nobody wants to get rid of `g_` from the code base more than me ;-) I'm in a bit of an hurry, and I'm not considering this "the Account C++ rewrite". Be comforted by the fact that I'm with you heart and soul on this point.